napalm_ansible/modules/napalm_get_config.py

```diff
@@ -0,0 +1,274 @@
+        description:
+          - Dictionary which acts as a collection of arguments used to define the characteristics
+            of how to connect to the device.
+            Note - hostname, username, password and dev_os must be defined in either provider
+            or local param
+            Note - local param takes precedence, e.g. hostname is preferred to provider['hostname']
+        required: False
+    dev_os:
+        description:
+          - OS of the device
+        required: False
+        choices: ['eos', 'junos', 'iosxr', 'fortios', 'ios', 'mock', 'nxos', 'nxos_ssh', 'panos',
+        'vyos']
+    timeout:
+        description:
+          - Time in seconds to wait for the device to respond
+        required: False
+        default: 60
+    optional_args:
+        description:
+          - Dictionary of additional arguments passed to underlying driver
+        required: False
+        default: None
+    dest:
+        description:
+            - File where to save retrieved configuration from a device. Configuration won't be
+              retrieved if not set.
+        default: None
+        required: True
+    type:
+        description:
+            - Config type to retreive from device. If retrieved config is empty retrieve 
+            "running" type instead.
+        default: running
+        required: False
+        choices: ['running', 'candidate', 'startup']
+    strip_comments:
+        description:
+            - Strip comments with timestamps from the config to behave in idempotent way. 
+        default: False
+        required: False
+'''
+
+EXAMPLES = '''
+- name: get the running config with stripped comments
+  napalm_get_config:
+    hostname: "{{ inventory_hostname }}"
+    username: "{{ username }}"
+    dev_os: "{{ os }}"
+    password: "{{ password }}"
+    dest: "../backup/{{ inventory_hostname }}"
+    type: running
+    strip_comments: True
+    
+- name: get the startup config using provider
+  napalm_get_config:
+    provider: "{{ ios_provider }}"
+    dest: "../backup/{{ inventory_hostname }}"
+    type: startup
+'''
+
+RETURN = '''
+changed:
+    description: whether the config was retrieved and saved to file
+    returned: always
+    type: bool
+    sample: True
+
+msg:
+    description: retrieved config file
+    returned: always
+    type: string
+    sample: "## Last commit: 2018-05-30 13:52:39 UTC by root ..."
+'''
+
+napalm_found = False
+try:
+    from napalm import get_network_driver
+    napalm_found = True
+except ImportError:
+    pass
+
+# Legacy for pre-reunification napalm (remove in future)
+if not napalm_found:
+    try:
+        from napalm_base import get_network_driver   # noqa
+        napalm_found = True
+    except ImportError:
+        pass
+
+
+def save_to_file(content, filename):
+    f = open(filename, 'w')
+    try:
+        f.write(content)
+    finally:
+        f.close()
+
+
+def main():
+    os_choices = ['eos', 'junos', 'iosxr', 'fortios', 'ios', 'mock', 'nxos',
```

Comment:
I think we should remove this, there is the PR #132 where we are actually doing this for the rest.

Reply:
done, removed